feat(fe): category feature implementation

ki1111ua opened the pull request:
category feature implementation check

Lamaalmulla committed:
feat(fe): category feature implementation

hengrv committed:
Merge branch 'frontenddev' into category-feature

hengrv reviewed (approved):
This looks really great - I will test locally and check functionality.

Merge conflicts have been fixed

hengrv commented on src/app/_components/screens/product-screen.tsx:
```diff
@@ -190,9 +218,71 @@ const ProductScreen = function ProductScreen({
+  //const product = products[currentIndex]
+  const product = filteredProducts[currentIndex]
 
   const handleViewDetails = useCallback(() => {
```
these hooks are rendered conditionally if the `filteredProducts` length is zero, and need to be moved above the early return inside the `useEffect`

hengrv reviewed (changes requested):
There's one small issue with some hooks being conditionally rendered which I can quickly fix

hengrv committed:
fix(fe): fix hook rendering order

some hooks had the potential to conditionally render after some code was introduced - moves them to always render

also fixed an issue with product potentially not being defined

hengrv commented:
Hook rendering order is fixed now, and deployment issues regarding a potentially undefined variable have been fixed

hengrv reviewed (approved):
Hook rendering order is fixed now, and deployment issues regarding a potentially undefined variable have been fixed
